### PR TITLE
add support for lint and test locally

### PR DESCRIPTION
**Pull request by giubacc**
# Describe your changes

add support for lint and test locally

## Issue ticket number and link

Fixes: https://github.com/aquarist-labs/s3gw/issues/677

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] CHANGELOG.md has been updated should there be relevant changes in this PR.

### CODE REVIEW

**Comment by jecluis** [comment]
@m-ildefons mind taking a look at this one? should we keep it, close it, leave it in the back burner?

**Comment by giubacc** [comment]
please let's reconsider if it is worth to keep this given that stand alone s3gw charts could have today a relative importance.

**Comment by jecluis** [comment]
> please let's reconsider if it is worth to keep this given that stand alone s3gw charts could have today a relative importance.

The Helm chart is not going anywhere. We just need to understand whether this is to be handled _now_ (i.e., for v0.23.0)

File: .ct/lintconf.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+---
```
Comment (m-ildefons) [inline .ct/lintconf.yaml:1]: Why not just use the defaults of `ct`?

File: .ct/lintconf.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  document-start: disable           # No --- to start a file
+  empty-lines:
+    max: 2
+    max-start: 0
+    max-end: 0
+  hyphens:
+    max-spaces-after: 1
+  indentation:
+    spaces: consistent
+    indent-sequences: whatever      # - list indentation will handle both indentation and without
```
Comment (m-ildefons) [inline .ct/lintconf.yaml:33]: No! Pick one and stick with it.

**Review by m-ildefons — CHANGES_REQUESTED** [review]
Pick an indent style.
Otherwise looks good to go.

File: .ct/lintconf.yaml (hunk 1 above)
Comment (giubacc) [reply .ct/lintconf.yaml:1]: I don't remember the exact reason why I had to add this, probably something with the added stuff broke the lint.
If you prefer, we can remove this and see what fails and see if that can be solved without this file.

File: .ct/lintconf.yaml (hunk 2 above)
Comment (m-ildefons) [reply .ct/lintconf.yaml]: I personally prefer having lists indented, so `true` would be required for this setting then.

Comment (giubacc) [reply .ct/lintconf.yaml]: @m-ildefons changed to `true`, this config should be taken from both local and remote lint; so we should be fine.

**Comment by jecluis** [comment]
@m-ildefons are you happy with the changes?

**Review by m-ildefons — APPROVED** [review]
lgtm

**Pull request merged**